### ISSUE TITLE
[New rule] avoid-non-ascii-symbols

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#819]
**Please describe what the rule should do:**
Hi thanks for the Flutter-Favorite package! I am proposing a new rule that enforces i18n/l10n. The basic idea is, search non-ascii characters (`[^\x00-\x7F]` in regex) that appears in string literals, and forbid that. This is because, for non-English apps, the text that needs l10n is usually non-ascii, so this rule efficiently finds out such text which need l10n. As for why only searching non-ascii characters, it is because there do exist many string literals which are quite reasonable and is not meant to be read by a user and should not do l10n. 

**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

```dart
Text('hello');
Text('hello 汉字'); // LINT
```

**Are you willing to submit a pull request to implement this rule?**

**Comment by incendial** [comment]
@fzyzcjy Am I right, that we can call this rule as "avoid-ascii-symbols" or something? And can you please fill examples sections just to make sure that I got your idea correctly

**Comment by incendial** [comment]
Available in 4.15.0 🚀

**Comment by fzyzcjy** [comment]
🎉
